metrics/cli/docs/features/doctor_output_improvements/diagrams/doctor_output_improvements_class_diagram.puml

```diff
@@ -0,0 +1,129 @@
+@startuml doctor_output_improvements_class_diagram
+
+package cli.lib as cli_lib {
+    package cli {
+        package command {
+            class DoctorCommand {
+                + name: String
+                + doctorFactory: DoctorFactory
+                + validationResultPrinter: ValidationResultPrinter
+
+                + run(): Future<void>
+            }
+        }
+
+        package doctor {
+            class Doctor {
+                - _coolService: CoolService
+                + validationResultBuilder: ValidationResultBuilder
+
+                + checkVersions(): Future<ValidationResult>
+            }
+        }
+    }
+
+    package services {
+        package cool_service {
+            package adapter {
+                class CoolCliServiceAdapter {
+                    - _versionHelper: VersionHelper
+                    - _coolServiceCli: CoolServiceCli
+
+                    + checkVersion(): Future<TargetValidationResult>
```

Comment:
🔨
```suggestion
                    + checkVersion(): Future<ProcessResult>
```
Also, should we remove the dependency between this class and the `TargetValidationResult`?